qcore/timeseries.py

```diff
@@ -445,8 +445,9 @@ def __init__(self, outbin):
         # protect against duplicated stations between processes
         # results in too many stations entries created, last ones are empty
         # important to keep indexes correct, only remove empty items from end
-        if stations.name[-1] == "":
-            stations = stations[: -np.argmin((stations.name == "")[::-1])]
+        if stations.name[-1] in ["", b""]:
+            stations = stations[: -np.argmin((stations.name == stations.name[-1])[::-1])]
+
```

Comment:
I'd prefer either `not stations.name[-1]` or `len(stations.name[-1]) == 0` as it catches both without the `in` keyword.

Reply:
Unless stations.name can be `None`?

Reply:
Interesting suggestion and an instant potential counterargument. 
It is unlikely for the stations name to be None. But I am not 100% sure. I prefer to keep this a bit more explicit with `""` and `b""`, as the code is already not easy to follow - took me a while to understand how we ran into this problem.

Reply:
yeah sure thing.

Reply:
To be honest I've been meaning to clean up that code entirely, it's quite difficult to understand.